cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -151,6 +151,45 @@ std::unique_ptr<KernelState> MinMaxInit(KernelContext* ctx, const KernelInitArgs
   return visitor.Create();
 }
 
+// ----------------------------------------------------------------------
+// All implementation
+
+struct BooleanAllImpl : public ScalarAggregator {
+  void Consume(KernelContext*, const ExecBatch& batch) override {
+    // short-circuit if seen a false already
+    if (this->all == false) {
+      return;
+    }
+
+    const auto& data = *batch[0].array();
+    arrow::internal::OptionalBinaryBitBlockCounter counter(
+        data.buffers[1], data.offset, data.buffers[0], data.offset, data.length);
+    int64_t position = 0;
+    while (position < data.length) {
+      const auto block = counter.NextOrNotBlock();
```

Comment:
Just wondering but should it in theory not be a "AndNot" instead of "OrNot" ?

Reply:
I named it after the bit operation we're using, which in this case is 
https://github.com/apache/arrow/blob/7efb373ace6489dbebf2367dd91b05208e48a1bb/cpp/src/arrow/util/bit_block_counter.h#L196-L197